### ISSUE TITLE
Savestates in Challenge Mode can be used to skip levels

### ISSUE THREAD

**Issue by ComplexPlane** [ComplexPlane/ApeSphere#21]
Loading savestates post-goal in Challenge Mode can let the player skip levels similar to the Challenge Mode Any% glitch. The solution probably involves savestating the Challenge Mode progress struct.

**Comment by rehtropsmb** [comment]
There's generally a lot of weird behavior with savestates in challenge mode, especially with multiplayer (both players must be using the same monkey).

I believe I've gotten a crash messing with savestates in multiplayer before, but I can't remember exactly how. Not really urgent since people usually don't use savestates in challenge mode (much less in multiplayer lol) but documenting here so it isn't forgotten.

I was just messing around with it and was able to load a savestate during spin-in. The game put the ball and monkey at the location of the savestate, but I still spawned at the start with an empty ball.
![image](https://github.com/ComplexPlane/SMB2PracticeMod/assets/139207294/8d181e46-c736-49f3-8427-e484ddfea455)

**Comment by ComplexPlane** [comment]
Yeah savestates assume a single player. I don't know if it's as easy as saving more than one ball/ape struct and the player count/current player but maybe it is.
